azurerm/resource_arm_app_service_custom_hostname_binding_test.go

```diff
@@ -0,0 +1,134 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"os"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func testAccAzureRMAppServiceCustomHostnameBinding_basic(t *testing.T) {
+	appServiceEnvVariable := "ARM_TEST_APP_SERVICE"
+	appServiceEnv := os.Getenv(appServiceEnvVariable)
+	if appServiceEnv == "" {
+		t.Skipf("Skipping as %q is not specified", appServiceEnvVariable)
+	}
+
+	domainEnvVariable := "ARM_TEST_DOMAIN"
+	domainEnv := os.Getenv(domainEnvVariable)
+	if domainEnv == "" {
+		t.Skipf("Skipping as %q is not specified", domainEnvVariable)
+	}
+
+	resourceName := "azurerm_app_service_custom_hostname_binding.test"
+	ri := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMAppServiceCustomHostnameBinding_basicConfig(ri, location, appServiceEnv, domainEnv)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAppServiceCustomHostnameBindingDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceCustomHostnameBindingExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAppServiceCustomHostnameBindingDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).appServicesClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_app_service_custom_hostname_binding" {
+			continue
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		appServiceName := rs.Primary.Attributes["app_service_name"]
+		hostname := rs.Primary.Attributes["hostname"]
+
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		resp, err := client.GetHostNameBinding(ctx, resourceGroup, appServiceName, hostname)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testCheckAzureRMAppServiceCustomHostnameBindingExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
```

Comment:
`%q` here?

Reply:
in general we don’t for these since this is the resource name (eg “azurerm_resource_group”) rather than the name property of the resource so generally we just output those raw (since it means the resource is missing from the state) - but I can update this?